hera_corr_cm/hera_corr_cm.py

```diff
@@ -81,24 +76,61 @@ def _get_response(self, command, timeout=10):
             sent_message = json.loads(command)
         except:
             self.logger.error("Failed to decode sent command")
-        target_time = sent_message["time"]
+        # time is probably a float, but will cast as a str for exact comparison
+        target_time = str(sent_message["time"])
+        if not isinstance(target_time, bytes):
+            target_time = target_time.encode()
+
         target_cmd = sent_message["command"]
+        # there is some python 2 vs 3 tension here. Force bytes to be consistent
+        if not isinstance(target_cmd, bytes):
+            target_cmd = target_cmd.encode()
+        wait_time = time.time()
         # This loop only gets activated if we get a response which
         # isn't for us.
         while(True):
-            message = self.corr_resp_chan.get_message(timeout=timeout)
-            if message is None:
-                self.logger.error("Timed out waiting for a correlator response")
-                return
+            # sleep for 1s between each attempt
+            time.sleep(1)
+
+            command_status = self.r.hgetall("corr:cmd_status")
+            # bool of a dict is False if the dict is empty
+            if not bool(command_status):
+                # command dict was read while correlator was creating a new status
+                continue
+
             try:
-                message = json.loads(message["data"])
-            except:
-                self.logger.warning("Got a non-JSON message on the correlator response channel")
+                response = json.loads(command_status[b"args"])
```

Comment:
I don't think that our dictionary keys should be bytes, but I think there might be a way around this. According to StackOverflow, it's possible to handle the conversion automagically if we pass `decode_responses=True` as a keyword to our redis client when we initialize it:

https://stackoverflow.com/a/44032018

I know it's a bit outside the original scope of this PR, but it might be useful to do this throughout the repo. We can leave it off for now, but I'll make an issue to do this.

Reply:
I'll look into it, if this can handle the py2/py3 bytes Unicode junk I'll implement it.

Reply:
The decode seems to work well.